### PR TITLE
[REF] Utilise new hook in CiviCRM for modifiying UF FIelds and Crate …

### PR DESCRIPTION
**Pull request by seamuslee001**
…standalone functions that are changed from core for use by the batch update page and standardise the process of generating form output

Note for this to work you will need to make sure you test this out on latest master code

### CODE REVIEW

**Comment by JoeMurray** [comment]
@seamuslee001 could you summarize a bit what you're doing here? It seems like you are taking a copy of some core files into the extension, but not overriding. This will allow the extension to work without interfering with core or requiring us to upgrade it everytime the copy and pasted file is changed. But it will mean that we may miss issues when the base core file is changed. So there is still an issue with maintainability, just a different one that manifests more silently.

**Comment by JoeMurray** [comment]
I'm okay with general direction after discussion in scrum, with caveat that I am concerned about maintainability of the code repurposed from core.

**Comment by JoeMurray** [comment]
Should we merge and release?

**Comment by Edzelopez** [comment]
I think this is ready for release after a brief QA, can you please confirm @seamuslee001 ?
